cartoframes/context.py

```diff
@@ -1674,17 +1895,46 @@ def data_augment(self, table_name, metadata):
              'instead.', DeprecationWarning)
         return self.data(table_name, metadata, persist_as=table_name)
 
+    def _table_touch(self, table_name, dtypes, overwrite=False):
+        """Create an empty table with a specific schema"""
+        if overwrite:
+            self.delete(table_name)
+        schema = ', '.join(
+            'null::{0} as {1}'.format(_dtype2pg(t), c)
+            for c, t in dtypes.items()
+        )
+        query = 'SELECT {} LIMIT 0'.format(schema)
+        print(query)
+        resp = self._auth_send(
+            'api/v1/imports',
+            'POST',
+            params={'sql': query, 'table_name': table_name}
+        )
+        while True:
+            r = self._auth_send(
+                'api/v1/imports/{}'.format(resp['item_queue_id']),
+                'GET'
+            )
+            if r['state'] == 'complete':
+                break
+            elif r['state'] == 'failure':
+                raise CartoException(str(r))
+            time.sleep(0.5)
```

Comment:
I'd use an exponential backoff saturating at, say, 30s to be nicer with the API's. Same in a similar loop above.

Reply:
Good idea, thanks :)